rest/server_context.go

```diff
@@ -265,9 +265,13 @@ func (sc *ServerContext) PostUpgrade(preview bool) (postUpgradeResults PostUpgra
 		}
 
 		// Index cleanup
-		removedIndexes, err := database.RemoveObsoleteIndexes(preview)
-		if err != nil {
-			return nil, err
+		var removedIndexes = make([]string, 0)
+		gocbBucket, ok := base.AsGoCBBucket(database.Bucket)
+		if ok && gocbBucket.HasN1qlNodes() {
+			removedIndexes, err = database.RemoveObsoleteIndexes(preview)
+			if err != nil {
+				return nil, err
```

Comment:
For both this and the DDocs one, I don't think we should bail out. We've handled one case here, but they're still dependant on each other. Say there's an issue contacting the Views API (port blocked because it was never in use?), can you ever reach this code?